src/app/txRunner.js

```diff
@@ -65,25 +65,28 @@ TxRunner.prototype.execute = function () {
         })
       })
     } else {
-      self.web3.eth.estimateGas(tx, function (err, resp) {
+      self.web3.eth.estimateGas(tx, function (err, gasEstimation) {
         if (err) {
-          return callback(err, resp)
+          return callback(err, gasEstimation)
         }
-
-        if (resp > gasLimit) {
-          return callback('Gas required exceeds limit: ' + resp)
-        }
-
-        tx.gas = resp
-
-        var sendTransaction = self.personalMode ? self.web3.personal.unlockAccountAndSendTransaction : self.web3.eth.sendTransaction
-
-        sendTransaction(tx, function (err, resp) {
+        self.web3.eth.getBlock('latest', function (err, block) {
           if (err) {
-            return callback(err, resp)
-          }
+            return callback(err)
+          } else {
+            var blockGasLimit = Math.floor(block.gasLimit - block.gasLimit / 1024)
+            tx.gas = blockGasLimit < gasEstimation ? blockGasLimit : gasEstimation
```

Comment:
Ah I see, so practically this is just a workaround for our inability to tell that the estimate failed.